### ISSUE TITLE
[Bug?]: how to render *404 page?

### ISSUE THREAD

**Issue by huseeiin** [solidjs/solid-start#1574]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

when i prerender `/about` it creates a `about/index.html` file correctly. but when i prerender `"/*404"` it generates nothing.

### Steps to reproduce 🕹

```ts
import { defineConfig } from "@solidjs/start/config";

export default defineConfig({
  server: {
    preset: "cloudflare-pages",
    prerender: { routes: ["/*404"] },
  },
});
```

**Comment by doeixd** [comment]
I dont think this is possible with Nitro? I'm not sure this is a SolidStart problem?

**Comment by huseeiin** [comment]
> I dont think this is possible with Nitro? I'm not sure this is a SolidStart problem?

well, since solidstart uses nitro, it is a solidstart problem even if it originates from nitro

**Comment by mdynnl** [comment]
using the code provided with these versions
```
├── @solidjs/meta@0.29.4
├── @solidjs/router@0.13.6
├── @solidjs/start@1.0.2
├── solid-js@1.8.18
└── vinxi@0.3.12
```

# cloudflare-pages
<img width="730" alt="Screenshot 2024-07-06 at 05 57 59" src="https://github.com/solidjs/solid-start/assets/53917354/89a85b47-2841-4ce7-9fb5-80b2d8e0148e">
but you probably don't need it for this preset as workers can already serve a 404 page


# cloudflare-pages-static

```ts
import { defineConfig } from '@solidjs/start/config';

export default defineConfig({
  server: {
    preset: 'cloudflare-pages-static',
    prerender: {
      // crawlLinks: true, // required for `cloudflare-pages` preset
      autoSubfolderIndex: false,
      routes: ['/404'],
    },
  },
});

```

This might be what you want which also generates a redirect rule `/* /404.html 404` which for some reason
[is not supported yet](https://developers.cloudflare.com/pages/configuration/redirects/#advanced-redirects:~:text=No,404.html%20404)
But `404.html` will be detected and will be used as 404 redirection expcept for `/404` which returns 200

<img width="815" alt="Screenshot 2024-07-06 at 06 27 31" src="https://github.com/solidjs/solid-start/assets/53917354/f692cc77-443c-428c-84e7-4a4c4efe5ae0">

**Comment by ryansolid** [comment]
I would expect you to be able to put `/404` and have it work because it doesn't exist. The thing is 404 page doesn't exist so to make it exist you need to put it in the list. And it should match the star route. From there you can set up your redirect rules in your provider to point to that page.
